src/EditorFeatures/Core/Implementation/FindReferences/AbstractFindReferencesService.cs

```diff
@@ -51,36 +59,125 @@ protected AbstractFindReferencesService(IEnumerable<IReferencedSymbolsPresenter>
             return null;
         }
 
-        public async Task<IEnumerable<INavigableItem>> FindReferencesAsync(Document document, int position, IWaitContext waitContext)
+        /// <summary>
+        /// Finds references using the externally defined <see cref="IFindReferencesResultProvider"/>s.
+        /// </summary>
+        private async Task AddExternalReferencesAsync(Document document, int position, ArrayBuilder<INavigableItem> builder, CancellationToken cancellationToken)
+        {
+            // CONSIDER: Do the computation in parallel.
+            foreach (var provider in _externalReferencesProviders)
+            {
+                var references = await provider.FindReferencesAsync(document, position, cancellationToken).ConfigureAwait(false);
+                if (references != null)
+                {
+                    builder.AddRange(references.WhereNotNull());
+                }
+            }
+        }
+
+        /// <summary>
+        /// Finds references using <see cref="SymbolFinder.FindReferencesAsync(ISymbol, Solution, CancellationToken)"/>
+        /// </summary>
+        private async Task AddSymbolReferencesAsync(Document document, int position, ArrayBuilder<INavigableItem> builder, IWaitContext waitContext)
         {
-            var cancellationToken = waitContext.CancellationToken;
             var result = await this.FindReferencedSymbolsAsync(document, position, waitContext).ConfigureAwait(false);
-            if (result == null)
+            if (result != null)
             {
-                return SpecializedCollections.EmptyEnumerable<INavigableItem>();
+                var referencedSymbols = result.Item1;
+                var searchSolution = result.Item2;
+
+                var q = from r in referencedSymbols
+                        from loc in r.Locations
+                        select NavigableItemFactory.GetItemFromSymbolLocation(searchSolution, r.Definition, loc.Location);
+
+                builder.AddRange(q);
             }
+        }
 
-            var referencedSymbols = result.Item1;
-            var searchSolution = result.Item2;
+        public async Task<IEnumerable<INavigableItem>> FindReferencesAsync(Document document, int position, IWaitContext waitContext)
+        {
+            var cancellationToken = waitContext.CancellationToken;
 
-            var q = from r in referencedSymbols
-                    from loc in r.Locations
-                    select NavigableItemFactory.GetItemFromSymbolLocation(searchSolution, r.Definition, loc.Location);
+            var builder = ArrayBuilder<INavigableItem>.GetInstance();
```

Comment:
is this mean we can't show results from both solution and coby at the same time? like people do FAR on Exception in coby and then show result from both coby and current solution?